vmngclient/api/task_status_api.py

```diff
@@ -0,0 +1,123 @@
+        exit_statuses_ids = [OperationStatusId.SUCCESS.value, OperationStatusId.FAILURE.value]
+
+        reboot_action = RebootAction(session,devices)
+        reboot_action.execute()
+
+        # Keep asking for reboot status until it's not in exit_statuses (Failure or Success)
+        # or timeout is not achieved (3000s)
+        TaskStatus(session).wait_for_completed(5,3000,exit_statuses,exit_statuses_ids,reboot_action.action_id)
+
+    """
+
+    def __init__(self, session: vManageSession) -> None:
+        self.session = session
+        self.status: str = ''
+        self.status_id: str = ''
+        self.activity: List[str] = []
+
+    def wait_for_completed(
+        self,
+        sleep_seconds: int,
+        timeout_seconds: int,
+        action_id: str,
+        exit_statuses: List[OperationStatus] = [
+            OperationStatus.SUCCESS,
+            OperationStatus.FAILURE,
+        ],
+        exit_statuses_ids: List[OperationStatusId] = [
+            OperationStatusId.SUCCESS,
+            OperationStatusId.FAILURE,
+        ],
+        activity_text: str = '',
+        action_url: str = '/dataservice/device/action/status/',
+    ) -> "TaskStatus":
+        """
+        Method to check action status
+
+        Args:
+            sleep_seconds (int): interval between action status requests
+            timeout_seconds (int): After this time, function will stop requesting action status
+            exit_statuses (Union[List[OperationStatus], str]): actions statuses that cause stop requesting action status
+            exit_statuses_ids (Union[List[OperationStatusId], str]): actions statuses ids
+             that cause stop requesting action status id
+            action_id (str): inspected action id
+            activity_text (str): activity text
+            action_url (str, optional): Action url. Defaults to '/dataservice/device/action/status/'
+
+        Returns:
+            bool: True if condition is met
+        """
+        exit_statuses = [cast(OperationStatus, exit_status.value) for exit_status in exit_statuses]
+        exit_statuses_ids = [cast(OperationStatusId, exit_status_id.value) for exit_status_id in exit_statuses_ids]
+
+        def check_status(action_data) -> bool:
+            """
+            Function checks if condition is met. If so,
+            wait_for_completed stops asking for task status
+
+            Args:
+                status (str): status of task
+                status_id (str): status id of task
+                activity (str): activity text
+
+            Returns:
+                bool: False if condition is met
+            """
+
+            if (self.status in exit_statuses) and (self.status_id in exit_statuses_ids):
+                if not activity_text or activity_text in self.activity:
+                    return False
+            return True
+
+        def _log_exception(self) -> None:
+            logger.error("Operation status not achieved in given time")
+            return None
```

Comment:
return statement is unnecessary when using -> None annotation

Reply:
Fixed